src/gen_leader.erl

```diff
@@ -933,7 +934,7 @@ loop(#server{parent = Parent,
               _Msg when Debug == [] ->
                 handle_msg(Msg, Server, Role, E);
               _Msg ->
-                Debug1 = sys:handle_debug(Debug, {?MODULE, print_event},
+                Debug1 = sys:handle_debug(Debug, fun print_event/3,
```

Comment:
This is not exactly the same in the case of hot-code loading.

The original version would use the new print_event for each new event as soon as new code is loaded. The modified version will continue to use the original print_event function when new code is loaded. This means that it may continue doing so until the third version is loaded, after which the old function reference is removed and the loop crashes.

Reply:
I don't think it would be a significant loss to not be able to hot load here.  Is there a better way though of resolving the dialyzer warning?